lib/StripeResource.js

```diff
@@ -192,14 +197,49 @@ StripeResource.prototype = {
       callback.call(
         self,
         new Error.StripeConnectionError({
-          message: 'An error occurred with our connection to Stripe',
+          message: self._generateConnectionErrorMessage(requestRetries),
           detail: error,
         }),
         null
       );
     }
   },
 
+  _shouldRetry: function(res, numRetries) {
+    // Retry on conflict or rate limit error, if we have retries left.
+    if (res && (res.statusCode === 409 || res.statusCode === 429)) {
+      return numRetries < this._stripe.getMaxNetworkRetries();
+    }
+
+    // Retry on connection error, if we have retries left.
+    if (!res && (numRetries < this._stripe.getMaxNetworkRetries())) {
+      return true;
+    }
+
+    return false;
+  },
+
+  _getSleepTime: function(numRetries) {
+    var initialNetworkRetryDelay = this._stripe.getInitialNetworkRetryDelay();
+    var maxNetworkRetryDelay = this._stripe.getMaxNetworkRetryDelay();
+
+    // Apply exponential backoff with initialNetworkRetryDelay on the
+    // number of numRetries so far as inputs. Do not allow the number to exceed
+    // maxNetworkRetryDelay.
+    var sleepSeconds = Math.min(
+      initialNetworkRetryDelay * Math.pow(numRetries - 1, 2),
+      maxNetworkRetryDelay
+    );
+
+    // Apply some jitter.
+    sleepSeconds *= 0.5 * (1 + Math.random());
+
+    // Never sleep more than the max allowed.
+    sleepSeconds = Math.max(initialNetworkRetryDelay, sleepSeconds);
```

Comment:
The comment seems to say something quite different from what this code seems to do?

Reply:
Well the good news is the code is right, but that comment is totally wrong. It should be "Never sleep _less_ than the max allowed".

Reply:
err, you mean, "Never sleep less than the initial delay" ?